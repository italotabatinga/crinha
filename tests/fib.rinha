let fib = fn (n) => {
  if (n < 2) {
    n
  } else {
    fib(n - 1) + fib(n - 2)
  }
};

print(fib(10))
print(fib(15))

let fibTco = fn (n, a, b) => {
  if (n == 0) {
    a
  } else {
    if (n == 1) {
      b
    } else {
      fibTco(n - 1, b, a + b)
    }
  }
};

print(fibTco(45, 0, 1))
print(fibTco(46, 0, 1))

let fibBug = fn (n) => {
    let v = n();
    if (v < 2) {
        v
    } else {
        fibBug(fn () => v - 1) + fibBug(fn () => v - 2)
    }
};
print(fibBug(fn () => 10))