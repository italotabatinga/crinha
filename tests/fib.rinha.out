55
610
1134903170
1836311903
55
